programs/manifest/src/state/market.rs

```diff
@@ -708,6 +708,20 @@ impl<Fixed: DerefOrBorrowMut<MarketFixed>, Dynamic: DerefOrBorrowMut<[u8]>>
             })?;
 
             if did_fully_match_resting_order {
+                // Get paid for removing a global order.
+                if get_helper::<RBNode<RestingOrder>>(dynamic, current_order_index)
+                    .get_value()
+                    .get_order_type()
+                    == OrderType::Global
+                {
+                    let global_trade_accounts_opt: &Option<GlobalTradeAccounts> = if is_bid {
+                        &global_trade_accounts_opts[0]
+                    } else {
+                        &global_trade_accounts_opts[1]
+                    };
+                    remove_from_global(&global_trade_accounts_opt, &maker)?;
```

Comment:
somehow both the case where an order has been matched and has been cancelled sound very similar, it would make the code easier to follow if the names were more descriptive wrt to the way gas prepayment is handled

Reply:
They are the same. The gas payer of the current transaction gets the prepayment. That is stored on the global_trade_accounts_opt